x/crosschain/keeper/msg_add_to_outtx_tracker.go

```diff
@@ -0,0 +1,106 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	eth "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+// AddToOutTxTracker adds a new record to the outbound transaction tracker.
+// only the admin policy account and the observer validators are authorized to broadcast this message.
+func (k msgServer) AddToOutTxTracker(goCtx context.Context, msg *types.MsgAddToOutTxTracker) (*types.MsgAddToOutTxTrackerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	chain := k.zetaObserverKeeper.GetParams(ctx).GetChainFromChainID(msg.ChainId)
+	if chain == nil {
+		return nil, observertypes.ErrSupportedChains
+	}
+	adminPolicyAccount := k.zetaObserverKeeper.GetParams(ctx).GetAdminPolicyAccount(observertypes.Policy_Type_group1)
+	isAdmin := msg.Creator == adminPolicyAccount
+
+	isObserver, err := k.zetaObserverKeeper.IsAuthorized(ctx, msg.Creator, chain)
+	if err != nil {
+		ctx.Logger().Error("Error while checking if the account is an observer", err)
+	}
+
+	isProven := false
+	if !(isAdmin || isObserver) && msg.Proof != nil {
+		txx, err := k.VerifyProof(ctx, msg.Proof, msg.BlockHash, msg.TxIndex, msg.ChainId)
+		if err != nil {
+			return nil, types.ErrCannotVerifyProof.Wrapf(err.Error())
+		}
+		err = k.VerifyOutTxTrackerProof(ctx, txx, msg.Nonce)
+		if err != nil {
+			return nil, types.ErrCannotVerifyProof.Wrapf(err.Error())
+		}
+		isProven = true
+	}
+
+	tracker, found := k.GetOutTxTracker(ctx, msg.ChainId, msg.Nonce)
+	hash := types.TxHashList{
+		TxHash:   msg.TxHash,
+		TxSigner: msg.Creator,
+	}
+	if !found {
+		k.SetOutTxTracker(ctx, types.OutTxTracker{
+			Index:    "",
+			ChainId:  chain.ChainId,
+			Nonce:    msg.Nonce,
+			HashList: []*types.TxHashList{&hash},
+		})
+		return &types.MsgAddToOutTxTrackerResponse{}, nil
+	}
+
+	var isDup = false
+	for _, hash := range tracker.HashList {
+		if strings.EqualFold(hash.TxHash, msg.TxHash) {
+			isDup = true
+			if isProven {
+				hash.Proved = true
+				k.SetOutTxTracker(ctx, tracker)
+				k.Logger(ctx).Info("Proof'd outbound transaction")
+				return &types.MsgAddToOutTxTrackerResponse{}, nil
+			}
+			break
+		}
+	}
+	if !isDup {
+		if isProven {
+			hash.Proved = true
+			tracker.HashList = append([]*types.TxHashList{&hash}, tracker.HashList...)
+			k.Logger(ctx).Info("Proof'd outbound transaction")
+		} else {
+			tracker.HashList = append(tracker.HashList, &hash)
+		}
+		k.SetOutTxTracker(ctx, tracker)
+	}
+	return &types.MsgAddToOutTxTrackerResponse{}, nil
+}
+
+func (k Keeper) VerifyOutTxTrackerProof(ctx sdk.Context, txx ethtypes.Transaction, nonce uint64) error {
```

Comment:
I created two functions `ValidateEVMOutTxBody` and `ValidateBTCOutTxBody` to verify the `msg` `tss` against the original transaction body to prevent data mismatch in [Bitcoin header N proof PR](https://github.com/zeta-chain/node/pull/1263/files) in the file x/crosschain/keeper/keeper_out_tx_tracker.go. We'll have to unify the workflow a bit for both chains.
The `ChainId`, `Nonce`, `txHash`, `sender` in message will be checked in there to make sure reporter is not lying.